### PR TITLE
DIG-1502: Opa implements user-specific authorizations

### PR DESCRIPTION
**Pull request by daisieh**
Candigv2-authx will implement the lists for pending_users and user authorizations in vault. Assuming that those are already there, this PR does the following on the Opa side:

* Gets the user's specific program authorization lists from Vault, calculates which programs the user is authorized to view, and allows the user to read programs they are either team members of or are specifically authorized to view.
* Adds DELETE operations as separately-authorized paths. Users authorized to curate can also delete.
* Adds tests for all of this.

Initialization will also create the empty pending_users list if it doesn't already exist.

Also instead of using the running instance's CLIENT_ID, we should be able to check the one in the `aud` that we've picked up when registering a federated server.

BONUS: There is now a github action that runs pytest automatically.

### CODE REVIEW

File: initialize_vault_store.py [hunk 1]
```diff
@@ -43,5 +43,12 @@
     print(str(e))
     sys.exit(4)
 
+# initialize pending users
```
Comment (kcranston) [inline initialize_vault_store.py:46]: in this scenario, does "pending" = waiting to be authorized to see anything at all (i.e. part of the user enrollement flow), or does it mean "has made a data access request and is waiting for approval / denial" (i.e. part of the DAC workflow)?

**Comment by kcranston** [comment]
Couple of documentation notes:
* Can you update the `permissions_engine/readme.md` with the changes? 
* Either add some text to the top-level readme that points to the other (or we could merge them and just have one in the root dir)
* documentation somewhere about what endpoints are available, and that many (most?) operation goes through the authz library, not OPA directly

**Comment by daisieh** [comment]
I updated and moved the readme. The endpoints are in ingest, as documented in the readme, not in authx.

File: initialize_vault_store.py (hunk 1 above)
Comment (daisieh) [reply initialize_vault_store.py:46]: The former. I have not implemented anything about DAC requests and how they'd be queued before approval.

Comment (daisieh) [reply initialize_vault_store.py:46]: I assume that if I did, they'd be called something more like "pending dac requests" or something.

**Review by kcranston — COMMENTED** [review]
LGTM now, merging

**Pull request merged**